LibSerialize.lua

```diff
@@ -1346,6 +1464,26 @@ function LibSerialize:DeserializeValue(input)
     return unpack(output, 1, outputSize)
 end
 
+function LibSerialize:DeserializeValueAsync(input, callback, opts)
```

Comment:
I don't think a WoW specific API is needed here.
Better to put it in a example code